D2IQ-86493 feat: manually attach cluster (#4202)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.2/clusters/attach-cluster/manually-attach-cluster/index.md

---
layout: layout.pug

> navigationTitle: Manually attach Cluster created with CLI

Reviewer: ```suggestion
navigationTitle: Manually attach a Cluster created with CLI
```

> title: Attach a Cluster manually created with CLI

Reviewer: ```suggestion
title: Manually attach a Cluster created with CLI
```

menuWeight: 50

> excerpt: Attach a Cluster manually created with CLI

Reviewer: ```suggestion
excerpt: Attach a Cluster that was created with the CLI manually
```

---

> When creating a cluster in a Workspace namespace using the DKP CLI it will not be attached automatically. To automatically attach a cluster you can generate the cluster objects using the DKP CLI `--dry-run -o yaml` flags and then using the [Advanced Creation of Konvoy Clusters][create_cluster_advanced].

Reviewer: ```suggestion
When you create a cluster in a Workspace namespace using the DKP CLI, it does not attach automatically. To automatically attach a cluster, generate the cluster objects using the DKP CLI `--dry-run -o yaml` flags and create a cluster following the instructions in the [Advanced Creation of Konvoy Clusters][create_cluster_advanced] guide.
```

Reviewer: Would this solve the problem for good? If a user does this, he/she won't have to manually attach clusters created afterwards? If yes, we can take my suggestion to make that a bit more clear. 
If this is a step the user needs to follow every time, we could consider including this info in the steps.

Author: This is the alternative, if you generate the yaml using the CLI with `--dry-run` then use the UI to create the cluster, it **will** automatically attach.

These steps are only needed if you use the CLI to create the cluster. And this won't be required in 2.3.

Reviewer: Awesome, thank you @TattdCodeMonkey . In that case, I think we could use the following. Does it sound good?
```suggestion
When you create a cluster in a Workspace namespace using the DKP CLI, it does not attach automatically. To automatically attach a cluster, generate the cluster objects using the DKP CLI `--dry-run -o yaml` flags and create a cluster as stated in the [Advanced Creation of Konvoy Clusters][create_cluster_advanced] guide.
```

> If the Cluster is created using the CLI it will appear in the UI as it is being Provisioned. It will then be in an Unattached state when Provisioning is completed.

Reviewer: ```suggestion
If you created the cluster using the CLI, it will appear in the UI as Provisioned. Once provisioning is completed, the state changes to Unattached.
```

Author: @gvasquezvargas the changes you made are not completely accurate.

> it will appear in the UI as Provisioned.

The status will go `Pending` -> `Provisioning` -> `Unattached` in this scenario, the user will never see the `Provisioned` status.

>  Once provisioning is completed, the state changes to Unattached.

I think `status` is preferable to `state` here, IMO.

Reviewer: Oh, I see! Thank you for clarifying that @TattdCodeMonkey ! Maybe this one is better? 
```suggestion
When you create a cluster using the CLI, you will be able to see it in the UI while it is being provisioned. Once provisioning is completed, the status will change to Unattached.
```

> ## Manually attach Cluster created with CLI

Reviewer: ```suggestion
## Manually attach a cluster that was created with the CLI
```

> From the CLI find the `name` of the `Cluster` created so that it can be referenced.

Reviewer: ```suggestion
From the CLI find out the `name` of the created `Cluster` so you can reference it later:
```

```bash
$ kubectl -n <workspace_namespace> get clusters
```

> Attach the Cluster by creating a `KommanderCluster` from the CLI.

Reviewer: ```suggestion
Attach the cluster by creating a `KommanderCluster`:
```
